File: binary_search.py
def binarySearch(upper_bound, num):
	tries = 0
	lower_bound = 1
	upper_bound = upper_bound
	while True:
		tries += 1
		guess = (lower_bound + upper_bound) / 2
		if guess == num:
			print 'You guessed my number in', tries, 'tries!'
			break
		elif guess > num:
			print guess, 'is too high.'
			upper_bound = guess - 1
		else:
			print guess, 'is too low.'
			lower_bound = guess + 1

if __name__ == '__main__':
	binarySearch(300, 22)

File: binarySearch.js
var binarySearch = function(pool, num) {
	var min = 1;
	var max = pool;
	var tries = 1;
	while (guess != num) {
		var guess = Math.floor((min + max) / 2);
		tries++;
		if (guess === num) {
			console.log("You guessed my number, " + guess + ", in " + tries + " tries!");
			break;
		}
		else if (guess > num) {
			console.log('Your guess ' + guess + ' is too high.');
			max = guess - 1;
		} else {
			console.log('Your guess ' + guess + ' is too low.');
			min = guess + 1;
		}
	
	}
};

if (require.main === module) {
    binarySearch(120, 34);
}